src/devices_models/devices/common/duration_constraints.jl

```diff
@@ -268,10 +268,10 @@ function device_duration_parameters!(
             for i in (t - duration_data[ix].up + 1):t
```

Comment:
```suggestion
            for i in UnitRange{Int}(t - duration_data[ix].up + 1, t)
```

This resolves the problem without that many type conversions. Also, we have to be careful here if there are fractional hour time constraints how are we handling it.